sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/patch/PatchOperationCore.java

```diff
@@ -13,27 +13,47 @@
 public final class PatchOperationCore<T> extends PatchOperation {
 
     private final String path;
+    private final String from;
     private final T resource;
 
     /**
      * Initializes a new instance of the {@link PatchOperationCore} class.
      *
      * @param operationType Specifies the type of Update operation
      * @param path          Specifies the path to target location.
-     * @param value         Specifies the value to be used
+     * @param value         Specifies the value to be used. In case of move operation it will be a string in the format
+     *                      of a path
      */
     public PatchOperationCore(PatchOperationType operationType, String path, T value) {
         super(operationType);
 
+        if (!(value instanceof String)) {
```

Comment:
why the `value` type needs to be string? For other patch operations, it could be object etc ?

Reply:
Necause value is the target json-path to the target node

Reply:
that is **path** right?

Reply:
Correct, this check should only take place if operationtype is Move.